aiohttp/client_proto.py

```diff
@@ -80,41 +84,70 @@ def is_connected(self) -> bool:
     def connection_lost(self, exc: Optional[BaseException]) -> None:
         self._drop_timeout()
 
-        if exc is not None:
-            set_exception(self.closed, exc)
-        else:
+        original_connection_error = exc
+        reraised_exc = original_connection_error
+
+        connection_closed_cleanly = original_connection_error is None
+
+        if connection_closed_cleanly:
             set_result(self.closed, None)
+        else:
+            assert original_connection_error is not None
```

Comment:
This looks like its only for typing. Usually I guard these with `TYPE_CHECKING` but thats just personal preference as I don't think anyone runs with `-O` here

Reply:
I think it'd be preferable to just recommend `-O` in production environments so it has no impact. We have asserts like this elsewhere in the code too.

Reply:
Yep. I think that sticking `TYPE_CHECKING` everywhere is a burden we don't want to introduce all over. It's indeed easier to recommend doing `-O`.

Reply:
Marking as resolved.